ground/src/main/java/com/google/android/ground/model/mutation/MediaMutation.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.model.mutation
+
+import java.util.Date
+
+/**
+ * Represents a local change to media associated with a submission.
+ *
+ * These mutations are directly dependent upon another mutation type: SubmissionMutation. At
+ * present, any MediaMutation will only have relevance in the context of a submission. To this end,
+ * several properties are derived from the mutation's associated submission mutation.
+ */
+data class MediaMutation(
+  override val type: Type,
+  val mediaId: String,
+  val submissionMutation: SubmissionMutation,
```

Comment:
This would require `SubmissionMutations` to always be loaded and attached when deserializing `MediaMutations`. Is that the intent? If not we may just want to pass the required fields here.

Reply:
Yes, my thinking here was that a number of the mutation fields on this type `loi ID`, `survey` etc should be derived from the corresponding submission. If we want these fields available on this type, I think this is how we should do it since it makes the derivation possible and the dependence clear.

Otherwise, we could simply omit these fields from this type, but it would require either refactoring `Mutation` a bit or moving this type out of the Mutation type tree.
 
I did have a  tentative change a while back that reduced `Mutation` to its truly common parts `id, type, syncstate, retrycount`.... we'd then define loiId etc. as fields on the subtypes. This sort of feels correct to me, so I'd be cool with doing it that way too. wdyt?

Reply:
`SubmissionMutation` can also be associated with an `LoiMutation`, but we don't actually represent that in the model. If you think it wouldn't be too much of a burden to deserialize the dependent mutation before deserializing  `MediaMutations` then this ok, but I think it will add some overhead in terms of impl.

Speaking of which, perhaps we should link `LoiMutation` to the `Submission` one somehow so they can be grouped/shown together in the sync status screen?